#166879999 Users should be able to rate articles

Manorlds-Eaglespark opened the pull request:
#### What does this PR do?
Users should be able to rate articles

#### Description of Task to be completed?
A 5-star rating system is the end goal. Let's have authenticated users get visual feedback of the average rating of their articles

#### How should this be manually tested?
Follow the setup steps on the [readme page](https://github.com/andela/ah-frontend-kronos) of this repo to set-up locally, you will notice star ratings on the articles and are able to rate when reading the article.

#### What are the relevant pivotal tracker stories?
[#166879999](https://www.pivotaltracker.com/story/show/166879999)

#### Screenshots (if appropriate)
<img width="732" alt="Screenshot 2019-07-26 at 10 38 45" src="https://user-images.githubusercontent.com/28473847/61934795-cfa00400-af91-11e9-9794-5dd0b2de303e.png">
![Uploading Screenshot 2019-07-26 at 10.39.59.png…]()

reifred commented on src/actions/articles/rating/rateArticleAction.js:
```diff
@@ -0,0 +1,42 @@
+import { ArticleConstants } from '../../actionTypes';
+
+const apiUrl = 'https://ah-backend-kronos-staging.herokuapp.com/api/articles/';
+const lastPart = '/rate';
```
I think we have url connections file that stores the url there is no need to declare the url again.

reifred commented on src/actions/articles/rating/updateRatingAction.js:
```diff
@@ -0,0 +1,33 @@
+import { ArticleConstants } from '../../actionTypes';
+
+const apiUrl = 'https://ah-backend-kronos-staging.herokuapp.com/api/articles/';
+const lastPart = '/rate';
```
The same at this point don't re-declare the url

reifred reviewed (changes requested):
Thanks @Manorlds-Eaglespark for the work done. However, I got an error after rating an article maybe you should consider correcting this.
![image](https://user-images.githubusercontent.com/22131999/61939467-024efa00-af9c-11e9-8bcd-4d139b895644.png)

Manorlds-Eaglespark commented:
> Thanks @Manorlds-Eaglespark for the work done. However, I got an error after rating an article maybe you should consider correcting this.
> ![image](https://user-images.githubusercontent.com/22131999/61939467-024efa00-af9c-11e9-8bcd-4d139b895644.png)
Ohh, I  forget to clean up resources properly after unmount. Thanks.

reifred commented:
> > Thanks @Manorlds-Eaglespark for the work done. However, I got an error after rating an article maybe you should consider correcting this.
> > ![image](https://user-images.githubusercontent.com/22131999/61939467-024efa00-af9c-11e9-8bcd-4d139b895644.png)
> 
> You could try refreshing your browser.

I did kindly look into it and notify me when it is solved

Manorlds-Eaglespark commented:
> > > Thanks @Manorlds-Eaglespark for the work done. However, I got an error after rating an article maybe you should consider correcting this.
> > > ![image](https://user-images.githubusercontent.com/22131999/61939467-024efa00-af9c-11e9-8bcd-4d139b895644.png)
> > 
> > 
> > You could try refreshing your browser.
> 
> I did kindly look into it and notify me when it is solved

Sure, alright.

Manorlds-Eaglespark commented:
> > > > Thanks @Manorlds-Eaglespark for the work done. However, I got an error after rating an article maybe you should consider correcting this.
> > > > ![image](https://user-images.githubusercontent.com/22131999/61939467-024efa00-af9c-11e9-8bcd-4d139b895644.png)
> > > 
> > > 
> > > You could try refreshing your browser.
> > 
> > 
> > I did kindly look into it and notify me when it is solved
> 
> Sure, alright.

I have done something to address this. Though it is not in the scope of this feature. Thanks.

reifred reviewed (approved):
Thanks for the good work done @Manorlds-Eaglespark

Manorlds-Eaglespark committed:
feature(article-rating): Users should be able to view and edit article rating

- users can view article ratings
- users are able to edit their article ratings.

[delivers #166879999]
